src/cbps_torch.py

```diff
@@ -139,38 +139,48 @@ def _covariate_differences(self, metric):
 
         return (std_diffs_unweight, std_diffs_weight)
 
-    def diagnose(self, method):
+    def diagnose(self, method, ax=None, scatter=False):
         """Plot standarized differences.
 
         Show plot of standarized differences for the balancing covariates to show
         the final balance achieved by the CBPS method.
 
         Args:
             method (str): method to calculate standarized differences
+            ax (matplotlib.axes, optional): axes to plot. Defaults to None.
+            scatter (bool, optional): Scatter plot of unweighted and weighted balance. Defaults to False.
 
         Returns:
             matplotlib.axes
         """
-        fig, ax = plt.subplots(figsize=(10, 6))
+        if not ax:
```

Comment:
added option to pass `ax` so that figures can be combined

Reply:
oh this is way nicer!